mesonbuild/envconfig.py

```diff
@@ -210,6 +211,14 @@ def get_cmake_use_exe_wrapper(self) -> bool:
         assert isinstance(res, bool)
         return res
 
+    def get_java_home(self) -> T.Optional[Path]:
+        value = self.properties.get('java_home')
+        if value is None:
+            value = os.environ.get('JAVA_HOME')
+            return Path(value) if value else None
```

Comment:
Is this uncessary? It looks like from the comment that java_home is already set from the environment before this.